docs/usage/middleware/creating-middleware.rst

```diff
@@ -110,13 +110,13 @@ explore another example - redirecting the request to a different url from a midd
 
        async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
            if Request(scope).session is None:
-               response = Redirect(path="/login")
+               response = ASGIRedirectResponse(path="/login")
```

Comment:
```suggestion
               response = Redirect(path="/login").to_asgi_response()
```
@provinzkraut I know you mentioned `ASGIRedirectResponse` would be the solution, but wouldn't this be better since `ASGIRedirectResponse` is a bit more low level?

Reply:
I believe you would also have to pass the app and request to `to_asgi_response`.

Reply:
This would actually have to be:

```python
request = Request(scope)
if request.session is None_
    response = Redirect(path="/login").to_asgi_response(app=None, request=request)
```
which IMO is a bit convoluted. I think it's fine to use the low-level responses in middlewares, since they are low-level as well. They're ASGI apps, so it makes sense to use ASGI constructs such as the low level responses.